src/main/java/com/onarandombox/MultiverseCore/commandtools/flags/FlagGroup.java

```diff
@@ -0,0 +1,132 @@
+package com.onarandombox.MultiverseCore.commandtools.flags;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * A group of flags.
+ */
+public class FlagGroup {
+    /**
+     * A builder for a flag group.
+     *
+     * @param name The name of the group.
+     * @return The builder.
+     */
+    public static @NotNull Builder builder(@NotNull String name) {
+        return new Builder(name);
+    }
+
+    private final String name;
+    private final List<MVFlag> flags;
+    private final List<String> keys;
+    private final Map<String, MVFlag> keysFlagMap;
+
+    /**
+     * Creates a new flag group.
+     *
+     * @param builder The builder.
+     */
+    protected FlagGroup(@NotNull Builder builder) {
+        name = builder.name;
+        flags = builder.flags;
+        keys = builder.keys;
+        keysFlagMap = builder.keysFlagMap;
+    }
+
+    /**
+     * Get the name of this group.
+     *
+     * @return The name of this group.
+     */
+    public @NotNull String getName() {
+        return name;
+    }
+
+    /**
+     * Get the flags contained within in this group.
+     *
+     * @return The flags in this group.
+     */
+    public @NotNull List<MVFlag> getFlags() {
+        return flags;
+    }
+
+    /**
+     * Get the keys of the flags contained within in this group. Does not include aliases.
+     *
+     * @return The keys of the flags in this group.
+     */
+    public @NotNull List<String> getKeys() {
+        return keys;
+    }
+
+    /**
+     * Get the keys and flags contained within in this group.
+     *
+     * @return The keys and flags in this group.
+     */
+    public @NotNull Map<String, MVFlag>getKeysFlagMap() {
+        return keysFlagMap;
+    }
```

Comment:
These expose mutable collections. Maybe consider if there's a way to just provide methods to do the needed things with the collections rather than returning the collections themselves?